File: app/src/main/java/com/example/marvel/view/RegisterActivity.java
package com.example.marvel.view;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.marvel.R;

import static android.widget.Toast.makeText;

public class RegisterActivity extends AppCompatActivity {

    Button save;
    EditText name, confirmpassword, email, password;
    Context context = this;


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);
        Window window = getWindow();
        window.setStatusBarColor(getColor(R.color.loginColor));
        getWindow().setNavigationBarColor(ContextCompat.getColor(this, R.color.loginColor));
        name = findViewById(R.id.username);
        email = findViewById(R.id.email);
        password = findViewById(R.id.password);
        confirmpassword = findViewById(R.id.confirm_password);
        save = findViewById(R.id.buttonSave);

        save.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (todosPreenchidos(name, email, password, confirmpassword)) {
                    Intent intent = new Intent(getApplicationContext(), HomeActivity.class);
                    startActivity(intent);
                } else {
                    makeText(context, "Please! Fill in all the fields.", Toast.LENGTH_LONG).show();
                }
            }

            private boolean todosPreenchidos(EditText... listaDeCamposDeTexto) {
                for (EditText campoDeTexto : listaDeCamposDeTexto) {
                    if (campoDeTexto.getText().toString().isEmpty()) return false;
                }
                return true;
            }
        });
    }
}

File: app/src/main/java/com/example/marvel/view/RegisterActivity.kt
package com.example.marvel.view

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.example.marvel.R
import com.example.marvel.viewmodel.ViewModelRegister
import com.google.firebase.auth.FirebaseAuth

class RegisterActivity : AppCompatActivity() {

    lateinit var firebaseAuth: FirebaseAuth
    lateinit var buttonSave: Button
    lateinit var username: EditText
    lateinit var password: EditText
    lateinit var confirmPassword: EditText
    lateinit var email: EditText

    private val viewModelRegister by lazy {
        ViewModelProviders.of(this).get(ViewModelRegister::class.java)
    }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        val window = window
        window.statusBarColor = getColor(R.color.loginColor)
        getWindow().navigationBarColor = ContextCompat.getColor(this, R.color.loginColor)

        buttonSave = findViewById(R.id.buttonSave)
        username = findViewById(R.id.username)
        password = findViewById(R.id.password)
        confirmPassword = findViewById(R.id.confirm_password)
        email = findViewById(R.id.email)

        viewModelRegister.validao.observe(this, Observer {
            if(it){
                startActivity(Intent(this, HomeActivity::class.java))
                Toast.makeText(this, R.string.welcome, Toast.LENGTH_SHORT).show()
            } else{
                Toast.makeText(this, R.string.fill_in_correctly, Toast.LENGTH_LONG).show()
            }
        })

        firebaseAuth = FirebaseAuth.getInstance()
        buttonSave.setOnClickListener(View.OnClickListener {
            viewModelRegister.validarCampo(username.text.toString(),email.text.toString(),password.text.toString(),confirmPassword.text.toString())
        })
    }
}